src/wallet/mod.rs

```diff
@@ -56,69 +59,56 @@ use crate::{
 mod cusf_block_producer;
 pub mod error;
 mod mine;
+mod mnemonic;
 
 type BundleProposals = Vec<(M6id, BlindedM6<'static>, Option<PendingM6idInfo>)>;
 
+type WrappedBdkWallet = Arc<Mutex<bdk_wallet::PersistedWallet<file_store::Store<ChangeSet>>>>;
+
 struct WalletInner {
     main_client: HttpClient,
     validator: Validator,
-    bitcoin_wallet: Mutex<bdk_wallet::PersistedWallet<file_store::Store<ChangeSet>>>,
+    // Unlocked, ready-to-go wallet: Some
+    // Locked wallet: None
+    bitcoin_wallet: RwLock<Option<WrappedBdkWallet>>,
```

Comment:
Why is there an RwLock over a type that contains an `Arc<Mutex<...>>`?
This looks like it should be
`Option<Mutex<bdk_wallet::PersistedWallet<file_store::Store<ChangeSet>>>>`

Reply:
I originally tried that (+ other variations over the same concept), and this was the only way I was able to get the type checker happy. The crux is in the parts where we assign to `self.bitcoin_wallet`, I couldn't get that to work without a `RwLock` + `Arc<Mutex<...>>`. If you're able to simplify the types, I'm very much open to that!

Reply:
Fixed in c948cd5459fee1a3e8a05be4c83dc17b9f65bc4c